content/en/docs/tutorials/configuration/updating-configuration-via-a-configmap.md

```diff
@@ -0,0 +1,381 @@
+configmap-env-var-59cfc64f7d-dpr98   1/1     Running   0          46s
+```
+
+The key-value pair in the ConfigMap is configured as an environment variable in the container of the Pod.
+Check this by viewing the logs of the Pod.
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+You should see an output similar to the below:
+```
+Found 3 pods, using pod/configmap-env-var-7c994f7769-l74nq
+Thu Jan  4 16:07:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:07:26 UTC 2024 The basket is full of apples
+```
+
+Edit the ConfigMap.
+```shell
+kubectl edit configmap fruits
+```
+
+In the editor that appears, change the value of key `fruits` from `apples` to `mangoes`. Save your changes.
+The kubectl tool updates the ConfigMap accordingly (if you see an error, try again).
+
+Here's an example of how that manifest could look after you edit it:
+```yaml
+apiVersion: v1
+data:
+  fruits: mangoes
+kind: ConfigMap
+# You can leave the existing metadata as they are.
+# The values you'll see won't exactly match these.
+metadata:
+  creationTimestamp: "2024-01-04T16:04:19Z"
+  name: fruits
+  namespace: default
+  resourceVersion: "1749472"
+
+```
+
+You should see the following output:
+```
+configmap/fruits edited
+```
+
+Tail the logs of the Deployment and observe the output for few seconds:
+```shell
+kubectl logs deployment/configmap-env-var
+```
+
+You will notice that the output remains unchanged:
+```
+Thu Jan  4 16:12:56 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:06 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:16 UTC 2024 The basket is full of apples
+Thu Jan  4 16:13:26 UTC 2024 The basket is full of apples
+
+```
+
+{{< note >}}
+Although the value of the key inside the ConfigMap has changed, the environment variable in the Pod still possesses the earlier value. This is because environment variables in a Pod are not updated until the Pod is rolled-over.
+{{< /note >}}
+
+Let's perform a rollout for the Deployment that will cause all the Pods to restart.
+```shell
+kubectl rollout restart deployment configmap-env-var
+```
+
+Check the deployment.
+```shell
+kubectl get deployment configmap-env-var
+```
+
+You should see an output similar to:
+```
+NAME                READY   UP-TO-DATE   AVAILABLE   AGE
+configmap-env-var   3/3     3            3           12m
+```
+
+Check the Pods.
+```shell
+kubectl get pods --selector=app=configmap-env-var
+```
+
+The older Pods will terminate and new ones will be created. You should finally see an output similar to:
+```
+NAME                                 READY   STATUS        RESTARTS   AGE
+configmap-env-var-6d94d89bf5-2ph2l   1/1     Running       0          13s
+configmap-env-var-6d94d89bf5-74twx   1/1     Running       0          8s
+configmap-env-var-6d94d89bf5-d5vx8   1/1     Running       0          11s
+```
+
+View the logs of the Deployment.
+```shell
+kubectl logs deployment/configmap-env-var
```

Comment:
```suggestion
# Pick one Pod that belongs to the Deployment, and view its logs
kubectl logs deployment/configmap-env-var
```

Reply:
This should not be necessary as kubectl logs deployments/configmap-volume picks one of the Pod at random and displays its logs.

Reply:
We're explaining what the command does, here.